### ISSUE TITLE
Getting this to work with Gitolite?

### ISSUE THREAD

**Issue by ITBlogger** [adrienthebo/puppet-git-hooks#9]
Hi guys,

This post receive hook has been indispensable to us in getting a git workflow built up for our Puppet modules. Much thanks in creating it and updating it with newer features like auto chown/chmod.

Now though we are trying to take things to the next step by using gitolite to control access to our repos at a more granular level.

I'm having issues getting this hook to work with gitolite (seems to be permission related) and was hoping that you guys had some insights/documentation on getting gitolite to work with git and puppet the same way that git and puppet currently work with this post-receive hook.

Thanks,

Alex

**Comment by ventz** [comment]
Alex,

Check out: https://github.com/ventz/pgsvr

This was my solution because I couldn't find anything that did all of those things.

**Comment by ITBlogger** [comment]
Hey thanks,

Yeah, I was hoping to put off learning R10K until we really needed to, heh.
Was hoping that would be at least two years off and not a month. :D

Regards,

Alex

**Comment by ventz** [comment]
r10K is amazing.

Honestly, there isn't that much "change" from an user perspective, and
the new iteration that was released recently (for the puppet
integration) makes it super easy to setup.

Basically, if you have dynamic environments with puppet, it can take you 15-20 minutes to setup r10k and understand what's going on. Really, it comes down to 3 parts: Your external modules (git or puppet forge), your Components (modules), and your Sites (profiles using the modules).

It's clean, and it really makes sense once you start using it. The benefits to the whole thing simply cannot be describe in words. It's truly elegant.

Anyway, if you need help, feel free to reach out.

-Ventz

**Comment by ITBlogger** [comment]
Yeah, so the problem seems to be that the gitolite user doesn't have permissions to run chown or chmod.

Most likely solution is to move the post receive hooks somewhere else like /usr/local/bin, give gitolite user permission in sudoers to run those hooks as root and have the post receive hooks in the gitolite folders just be a simple script that calls those using sudo "sudo /usr/local/bin/puppet-post-receive" for instance.
